### PR TITLE
ci: add CI settings

### PR DESCRIPTION
**Pull request by kenji-miyake**
A follow-up for https://github.com/autowarefoundation/.github/pull/1#issuecomment-1076206175.

### CODE REVIEW

File: .github/sync-files.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    - source: .markdown-link-check.json
+    - source: .markdownlint.yaml
+    - source: .pre-commit-config-optional.yaml
+    - source: .prettierignore
+    - source: .prettierrc.yaml
+    - source: .yamllint.yaml
+
+- repository: autowarefoundation/autoware_common
+  files:
+    - source: .github/workflows/sync-files.yaml
```
Comment (kenji-miyake) [inline .github/sync-files.yaml]: I'll add `sync-files.yaml` to autowarefoundation/autoware as well to keep this settings simple.

Comment (kenji-miyake) [reply .github/sync-files.yaml]: https://github.com/autowarefoundation/autoware/pull/110

**Review by xmfcx — APPROVED** [review]

**Comment by xmfcx** [comment]
I think it also needed a `.markdownlint.yaml` file.

**Comment by kenji-miyake** [comment]
Oh, sorry. I'll add it in https://github.com/autowarefoundation/.github/pull/4. -> already added.

**Pull request merged**